src/contour/Config.h

```diff
@@ -144,138 +150,332 @@ struct SimpleColorConfig
+        /// excluding US-ASCII glyphs, cursor shapes and decorations.
+        ///
+        /// Value must be at least as large as grid cells available in the current view.
+        /// This value is automatically adjusted if too small.
+        crispy::lru_capacity textureAtlasTileCount = crispy::lru_capacity { 4000 };
 
-// NB: All strings in here must be UTF8-encoded.
-struct Config
+        // Configures the size of the PTY read buffer.
+        // Changing this value may result in better or worse throughput performance.
+        //
+        // This value must be integer-devisable by 16.
+        size_t ptyReadBufferSize = 16384;
+
+        // Size in bytes per PTY Buffer Object.
+        //
+        // Defaults to 1 MB, that's roughly 10k lines when column count is 100.
+        size_t ptyBufferObjectSize = 1024lu * 1024lu;
+
+        bool reflowOnResize = true;
+
+        std::unordered_map<std::string, vtbackend::ColorPalette> colorschemes;
+        std::unordered_map<std::string, TerminalProfile> profiles;
+        std::string defaultProfileName;
+
+        TerminalProfile* profile(std::string const& name)
+        {
+            assert(!name.empty());
+            if (auto i = profiles.find(name); i != profiles.end())
+                return &i->second;
+            assert(false && "Profile not found.");
+            return nullptr;
+        }
+
+        TerminalProfile const* profile(std::string const& name) const
+        {
+            assert(!name.empty());
+            if (auto i = profiles.find(name); i != profiles.end())
+                return &i->second;
+            assert(false && "Profile not found.");
+            return nullptr;
+        }
+
+        TerminalProfile& profile() noexcept
+        {
+            if (auto* prof = profile(defaultProfileName); prof)
+                return *prof;
+            crispy::unreachable();
+        }
+        TerminalProfile const& profile() const noexcept
+        {
+            if (auto const* prof = profile(defaultProfileName); prof)
+                return *prof;
+            crispy::unreachable();
+        }
+
+        // selection
+        std::string wordDelimiters;
+        vtbackend::Modifier bypassMouseProtocolModifier = vtbackend::Modifier::Shift;
+        SelectionAction onMouseSelection = SelectionAction::CopyToSelectionClipboard;
+        vtbackend::Modifier mouseBlockSelectionModifier = vtbackend::Modifier::Control;
+
+        // input mapping
+        InputMappings inputMappings;
+
+        bool spawnNewProcess = false;
+        std::shared_ptr<logstore::sink> loggingSink;
+
+        bool sixelScrolling = true;
+        vtbackend::ImageSize maxImageSize = {}; // default to runtime system screen size.
+        unsigned maxImageColorRegisters = 4096;
+
+        std::set<std::string> experimentalFeatures;
+    };
+} // namespace old
+
+template <typename T>
+struct ConfigEntry
 {
-    std::filesystem::path backingFilePath;
+    using val_type = T;
```

Comment:
```suggestion
    using value_type = T;
```

I noticed this a bit more often, but only clicked on it _some_ times. But let's generally try to avoid abbreviations, unless there's a good reason :)